dotcom-rendering/scripts/webpack/utils.ts

```diff
@@ -0,0 +1,27 @@
+import type LoadablePlugin from '@loadable/webpack-plugin';
+import type {
+	WebpackPluginInstance,
+	Configuration,
+	DefinePlugin,
+	IgnorePlugin,
+} from 'webpack';
+import type { BundleAnalyzerPlugin } from 'webpack-bundle-analyzer';
+import type FilterWarningsPlugin from 'webpack-filter-warnings-plugin';
+import type WebpackMessages from 'webpack-messages';
+
+type ValidPlugin =
+	| WebpackPluginInstance
+	| DefinePlugin
+	| FilterWarningsPlugin
+	| LoadablePlugin
+	| IgnorePlugin
+	| BundleAnalyzerPlugin
+	| WebpackMessages;
+
+export const isWebpackPluginInstance = (
+	p: false | ValidPlugin,
+): p is WebpackPluginInstance => p !== false;
```

Comment:
what if the plugin doesn't have types, is this still true? what does this result in?

```ts
[Math.random()].filter(isWebpackPluginInstance)
```

Reply:
```
Type 'number' is not assignable to type 'false | ValidPlugin'
```